### PR TITLE
builtin: add builtin-function 'to_seconds'

### PR DESCRIPTION
**Pull request by spongedu**
Add implementation for builtin function 'TO_SECONDS'

### CODE REVIEW

File: expression/builtin_time.go [hunk 1]
```diff
@@ -2016,7 +2016,26 @@ type builtinToSecondsSig struct {
 
 // See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_to-seconds
 func (b *builtinToSecondsSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("TO_SECONDS")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	if args[0].IsNull() {
+		d.SetNull()
```
Comment (shenli) [inline expression/builtin_time.go]: d is already nil here.

File: expression/builtin_time.go [hunk 2]
```diff
@@ -2016,7 +2016,26 @@ type builtinToSecondsSig struct {
...
+		return d, nil
+	}
+	sc := b.ctx.GetSessionVars().StmtCtx
+	date, err := convertDatumToTime(sc, args[0])
+	if err != nil {
+		return d, errorOrWarning(err, b.ctx)
+	}
+	ret := types.TimestampDiff("SECOND", types.ZeroDate, date)
+	if ret == 0 {
+		d.SetNull()
```
Comment (shenli) [inline expression/builtin_time.go]: So as here.

File: expression/builtin_time.go (hunk 1 above)
Comment (spongedu) [inline expression/builtin_time.go]: OK.

File: expression/builtin_time.go (hunk 2 above)
Comment (spongedu) [inline expression/builtin_time.go]: ok.

**Comment by shenli** [comment]
LGTM
@spongedu Please resolve the conflicts.
@coocood @XuHuaiyu PTAL

**Comment by hanfei1991** [comment]
LGTM @spongedu please resolve conflicts again

**Pull request merged**